Bundle complete types

**Is your feature request related to a problem? Please describe.**
I have a directory setup using symlinks and usually use `tsconfig.json` `paths` like this:
```
"paths": {
      "three": ["./node_modules/three", "./node_modules/@types/three"],
      "three/webgpu": ["./node_modules/@types/three/src/Three.WebGPU.d.ts"],
```

For motion this currently looks like this:
```
      "motion": ["./node_modules/motion/types/index.d.ts"],
      "@motionone/types": ["./node_modules/@motionone/types/types/index.d.ts"],
      "@motionone/dom": ["./node_modules/@motionone/dom/types/index.d.ts"],
      "@motionone/animation": ["./node_modules/@motionone/animation/types/index.d.ts"],
      "@motionone/generators": ["./node_modules/@motionone/generators/types/index.d.ts"]
```
And this isn't even working:

```

1 export declare const animate: (elements: import("..").ElementOrSelector, keyframes: import("./types").MotionKeyframesDefinition, options?: import("./types").AnimationOptionsWithOverrides) => import("@motionone/types").AnimationControls;
                                                  ~~~~

node_modules/@motionone/types/types/MotionValue.d.ts:1:65 - error TS2792: Cannot find module './'. Did you mean to set the 'moduleResolution' option to 'nodenext', or to add aliases to the 'paths' option?

1 import type { AnimationGenerator, BasicAnimationControls } from "./";
```

**Describe the solution you'd like**
Just bundle complete type and don't scatter them across multiple packages for a simple reason: The code is also bundled...

That is: Either don't do a import of modules of `index.d.ts` or preferably build a `dist/index.d.ts` and include it in the package distributed by NPM.

**Describe alternatives you've considered**
Switching to a more friendly library

--- a/packages/framer-motion/package.json
+++ b/packages/framer-motion/package.json
@@ -6,7 +6,7 @@
     "module": "dist/es/index.mjs",
     "exports": {
         ".": {
-            "types": "./dist/types/index.d.ts",
+            "types": "./dist/index.d.ts",
             "require": "./dist/cjs/index.js",
             "import": "./dist/es/index.mjs",
             "default": "./dist/cjs/index.js"
@@ -30,7 +30,7 @@
             "default": "./dist/cjs/dom.js"
         },
         "./client": {
-            "types": "./dist/types/client.d.ts",
+            "types": "./dist/client.d.ts",
             "require": "./dist/cjs/client.js",
             "import": "./dist/es/client.mjs",
             "default": "./dist/cjs/client.js"
@@ -54,7 +54,7 @@
         },
         "./package.json": "./package.json"
     },
-    "types": "dist/types/index.d.ts",
+    "types": "dist/index.d.ts",
     "author": "Matt Perry",
     "license": "MIT",
     "repository": "https://github.com/motiondivision/motion/",

--- a/packages/framer-motion/rollup.config.mjs
+++ b/packages/framer-motion/rollup.config.mjs
@@ -163,18 +163,7 @@ export const es = Object.assign({}, config, {
 
 const typePlugins = [dts({compilerOptions: {...tsconfig, baseUrl:"types"}})]
 
-const types = {
-    input: ["types/index.d.ts", "types/client.d.ts"],
-    output: {
-        format: "es",
-        entryFileNames: "[name].d.ts",
-        dir: "dist",
-    },
-    plugins: typePlugins,
-}
-
-
-function createTypes(input, file) {   
+function createTypes(input, file) {
     return {
         input,
         output: {
@@ -185,7 +174,8 @@ function createTypes(input, file) {
     }
 }
 
-
+const indexTypes = createTypes("types/index.d.ts", "dist/index.d.ts")
+const clientTypes = createTypes("types/client.d.ts", "dist/client.d.ts")
 const miniTypes = createTypes("types/mini.d.ts", "dist/mini.d.ts")
 const debugTypes = createTypes("types/debug.d.ts", "dist/debug.d.ts")
 const animateTypes = createTypes("types/dom.d.ts", "dist/dom.d.ts")
@@ -207,7 +197,8 @@ export default [
     cjsDomMini,
     cjsM,
     es,
-    types,
+    indexTypes,
+    clientTypes,
     debugTypes,
     mTypes,
     miniTypes,

--- a/packages/framer-motion/scripts/check-bundle.js
+++ b/packages/framer-motion/scripts/check-bundle.js
@@ -1,11 +1,35 @@
 const path = require("path")
-const { readFileSync } = require("fs")
+const { existsSync, readFileSync } = require("fs")
+const pkg = require("../package.json")
 
-const file = readFileSync(
-    path.join(__dirname, "../", "dist", "dom.d.ts"),
-    "utf8"
-)
+const dist = path.join(__dirname, "..", "dist")
+
+const file = readFileSync(path.join(dist, "dom.d.ts"), "utf8")
 
 if (file.includes(`<reference types="react" />`)) {
     throw new Error("DOM bundle includes reference to React")
+}
+
+/**
+ * Verify every "types" entry in package.json exports points to a file that
+ * exists, and that the bundled .d.ts file is self-contained (no relative
+ * imports of internal chunks). Issue #2900.
+ */
+for (const [name, entry] of Object.entries(pkg.exports)) {
+    if (!entry || typeof entry !== "object" || !entry.types) continue
+
+    const typesPath = path.join(__dirname, "..", entry.types)
+    if (!existsSync(typesPath)) {
+        throw new Error(
+            `Types file for "${name}" (${entry.types}) does not exist`
+        )
+    }
+
+    const contents = readFileSync(typesPath, "utf8")
+    const relativeImport = contents.match(/from ['"](\.[^'"]+)['"]/)
+    if (relativeImport) {
+        throw new Error(
+            `Types file for "${name}" (${entry.types}) contains a relative import (${relativeImport[1]}) — types must be bundled into a single self-contained file`
+        )
+    }
 }

